airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_source.py

```diff
@@ -111,6 +113,23 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         except ValidationError as exc:
             raise ConfigValidationError(FileBasedSourceError.CONFIG_VALIDATION_ERROR) from exc
 
+    def read(
+        self,
+        logger: logging.Logger,
+        config: Mapping[str, Any],
+        catalog: ConfiguredAirbyteCatalog,
+        state: Optional[Union[List[AirbyteStateMessage], MutableMapping[str, Any]]] = None,
+    ) -> Iterator[AirbyteMessage]:
+        yield from super().read(logger, config, catalog, state)
+        # count streams using a certain parser
+        stream_by_parser: Dict[str, int] = defaultdict(lambda: 0)
+        parsed_config = self._get_parsed_config(config)
+        for stream in parsed_config.streams:
+            stream_by_parser[stream.format.filetype] += 1
+
+        for (parser, count) in stream_by_parser.items():
+            yield create_analytics_message(f"file-cdk-{parser}-stream-count", count)
```

Comment:
```suggestion
        parsed_config = self._get_parsed_config(config)
        for parser, count in Counter(stream.format.filetype for stream in parsed_config.streams).items():
            yield create_analytics_message(f"file-cdk-{parser}-stream-count", count)
```
(`Counter` is imported from the standard library's `collections` module.)